datumaro/plugins/coco_format/extractor.py

```diff
@@ -236,19 +274,37 @@ def __call__(self):
             return mask_utils.merge(rles)
 
     def _get_label_id(self, ann):
-        if not ann["category_id"]:
+        cat_id = self._parse_field(ann, "category_id", int)
+        if not cat_id:
             return None
-        return self._label_map[ann["category_id"]]
+
+        label_id = self._label_map.get(cat_id)
+        if label_id is None:
+            raise InvalidLabelError(str(cat_id))
+        return label_id
+
+    def _parse_field(
+        self, ann: Dict[str, Any], key: str, cls: Union[None, Type[T], Tuple[Type, ...]]
+    ) -> T:
+        value = ann.get(key, NOTSET)
+        if value is NOTSET:
+            raise MissingFieldError(key)
+        elif cls and not isinstance(value, cls):
+            cls = (cls,) if isclass(cls) else cls
```

Comment:
FWIW, `isclass(cls)` is just `isinstance(cls, type)`. I don't think there's much reason to use `isclass` in Python 3 (in Python 2 it made more sense, since there were several kinds of classes).

Reply:
Maybe, but I don't see this function being deprecated even in 3.10, so I think, it's totally legitimate. Also, I see it more expressive.